File: solutions/ruby/raindrops/1/raindrops.rb
class Raindrops
  # Definimos los sonidos en una constante para mantener el código limpio
  SOUNDS = { 3 => "Pling", 5 => "Plang", 7 => "Plong" }

  def self.convert(number)
    # 1. Recorremos el diccionario y seleccionamos los sonidos si el número es divisible
    result = SOUNDS.map { |divisor, sound| sound if (number % divisor).zero? }.join

    # 2. Si result está vacío, devolvemos el número como string; si no, devolvemos result
    result.empty? ? number.to_s : result
  end
end

File: solutions/r/raindrops/1/raindrops.R
raindrops <- function(number) {
  result <- ""
  
  # Check divisibility by 3
  if (number %% 3 == 0) {
    result <- paste0(result, "Pling")
  }
  
  # Check divisibility by 5
  if (number %% 5 == 0) {
    result <- paste0(result, "Plang")
  }
  
  # Check divisibility by 7
  if (number %% 7 == 0) {
    result <- paste0(result, "Plong")
  }
  
  # If no factors matched, return the number as a string
  if (result == "") {
    return(as.character(number))
  }
  
  return(result)
}